### ISSUE TITLE
wrong instruction generation

### ISSUE THREAD

**Issue by jnk0le** [riscv-non-isa/riscv-asm-manual#78]
in "GOT-indirect addressing"

>1:
	auipc	a0, %got_pcrel_hi(msg + 1)
	ld	a0, %pcrel_lo(1b)(a0)

>Which generates the following assembler output and relocations as seen by objdump:

>0000000000000000 <.text>:
   0:	00000517          	auipc	a0,0x0
			0: R_RISCV_GOT_HI20	msg+0x1
   4:	00050513          	mv	a0,a0
			4: R_RISCV_PCREL_LO12_I	.L1

`ld` should not expand into `mv`, looks like copypasted from section above

**Comment by pdfrod** [comment]
This doesn't seem fixed? I'm still seeing `mv  a0,a0` in [GOT-indirect addressing
](https://github.com/riscv-non-isa/riscv-asm-manual/blob/3b06fa93fd52f277e8e0dc180bbb2ceec1ff5957/src/asm-manual.adoc#got-indirect-addressing) section.

**Comment by aswaterman** [comment]
Agreed, that line should be replaced with `   4: 00053503            ld a0,0(a0)`.  The rest is correct.

**Comment by jnk0le** [comment]
one thing is that this could be actual intermediate output from compilers which gets "relaxed" at later stages.

**Comment by aswaterman** [comment]
Yeah, but not in this particular situation, AFAICS.
